dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/cookies/CookieUtils.java

```diff
@@ -546,11 +554,35 @@ private static String convertV1SetCookie2(HttpCookie cookie) {
             buffer.append(value);
         }
 
+        //Servlet 6.0
+        setAttributes(cookie, buffer);
+
         if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
             Tr.debug(tc, "Created v1 Set-Cookie2: [" + buffer.toString() + "]");
         }
 
         return buffer.toString();
     }
 
+    /*
+     * since Servlet 6.0 - Support Cookie's setAttribute
+     * exclude "samesite", "port", "commenturl" attribute since they already set
+     */
+    private static void setAttributes(HttpCookie cookie, StringBuilder buffer) {
+        Map<String, String> cookieAttrs = cookie.getAttributes();
+        if (cookieAttrs != null) {
+            String key, value;
+            for (Entry<String, String> entry : cookieAttrs.entrySet()) {
+                key = entry.getKey();
+                if (!(key.equals("samesite") || key.equals("port") || key.equals("commenturl"))) {
+                    value = entry.getValue();
+                    if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                        Tr.debug(tc, "setAttribute (" + key + " , " + value + ")");
+                    }
+                    buffer.append("; " + key + "="); //performance impact if we capitalize first letter att name?
```

Comment:
I think this is because the convention of all other attributes we parse is to have the first letter capitalized. Since all values are normalized into lower case when they are put into the list, there would need to be an extra step here to keep the convention for every attribute that is generically being obtained. It runs it once per every attribute per cookie per response, so I suppose Phu is concerned about the extra work being made for every response. I don't think this sort of operation would be too harmful, but it would be something to take to the performance team to ensure we don't
change our performance.

Reply:
As discussed, we will leave the attributes in all lower case since there is no functionality issue with that setting.